ABM_CE_PV_ConsumerAgents.py

```diff
@@ -922,6 +961,106 @@ def storage_management(self, limited_paths):
             self.number_new_prod_hoarded = 0
             limited_paths["hoard"] = False
 
+        self.update_product_storage_hazardous() 
+        if self.is_hazardous_waste_storage_limit_exceeded():
+            # This is to prevent
+            # double counting of hazardous products since
+            # number_product_hoarded_hazardous is included in
+            # number_product_hoarded
+            if self.product_storage_to_other == 0:
+                self.product_storage_to_other = self.number_product_hoarded_hazardous
+            self.number_product_hoarded_hazardous = 0
+            self.product_years_storage_hazardous = []
+            limited_paths["hoard"] = False
+
+    def update_product_storage_hazardous(self): 
+        """
+        Update the storage of hazardous products based on the purchase choice.
+        """
+        if self.hazardous:
+            # product is not hazardous if purchase choice is used
+            if self.purchase_choice == "new":
+                self.product_years_storage_hazardous.append(self.EoL_pathway)
+            elif self.purchase_choice == "used":
```

Comment:
Although a good idea I don't think that is what would happen. Or at least not only that. There are two issues I see:

- The "used" condition is for the **purchase** at time _t_. It does not indicate if the **waste** at time _t_ is going to be reused or not (which I think is what Taylor said (_if_ the PV owner sell its panel(s) for reuse then he is exempt of all the hazardous stuff).
- Rather than using self.purchase_choise == "new" or "used" I think you should use "self.EoL_pathway"  (i.e., if self.EoL_pathway == "reuse" then exempt).